firmware/library/L3_HAL/Piezo_Driver.hpp

```diff
@@ -0,0 +1,31 @@
+// This driver is a layout for the piezo buzzer interface
+
+#pragma once
+#include <cstdint>
+
+class BuzzerInterface
+{
+ public:
+        // This function is responsible for selecting the pin config
+        virtual void Init(uint8_t pin_number) = 0;
+
+        // This is responsible to change the volume
+        // of the buzzer
+        // The volume will be between 0.0 and 1.0
+        // 0.0 = 0% duty cycle
+        // 1.0 = 50% duty cycle (max volume)
+        virtual void SetVolume(float volume) = 0;
+
+        virtual void Stop() = 0;
+
+        // beep continuously for a certain period of time
+        // @ a specified frequency
+        virtual void Beep(uint32_t frequency, uint32_t duration) = 0;
+
+        // produce a sequence of beeps for a specified duration
+        // @ a specified frequency
+        virtual void ProduceSequence(uint32_t freq, uint32_t dur) = 0;
+
+        // produce a musical tone for a specified duration
+        virtual void MakeTone(uint32_t duration) = 0;
```

Comment:
I think that having the two functions, `Beep` and `MakeTone` is unnecessary. You can combine them into 1 
function. They do the same thing thus you can write the method declaration as:

    virtual void Beep(uint32_t duration, uint32_t frequency = 1'000)

This would eliminate the need for MakeTone. Or you could name it MakeTone and remove Beep. Which ever name you feel fits the best.

I'm still now sure what ProduceSequence does. Can you explain how it works?